src/igz_controls/services/validation.service.js

```diff
@@ -0,0 +1,579 @@
+                generateRule.validCharacters('0-9 + -'),
+                generateRule.length({min: 4, max: lengths.phone})
+            ]
+        };
+
+        return {
+            storage: /^[a-zA-Z0-9]+?:\/\/[a-zA-Z0-9_.-]+?:[a-zA-Z0-9_./-]+?@[a-zA-Z0-9_.-]+?$/,
+            url: /^[a-zA-Z0-9]+?:\/\/[a-zA-Z0-9_.-]+?:[a-zA-Z0-9_\-.]+?@[a-zA-Z0-9_.-]+?$/,
+            timestamp: /^(?:\d{4})-(?:0[1-9]|1[0-2])-(?:0[1-9]|[1-2]\d|3[0-1])T(?:[0-1]\d|2[0-3]):[0-5]\d:[0-5]\d(?:\.\d+)?((?:[+-](?:[0-1]\d|2[0-3]):[0-5]\d)|Z)?$/,
+            ip: /^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/,
+            mask: /^(((255\.){3}(255|254|252|248|240|224|192|128|0+))|((255\.){2}(255|254|252|248|240|224|192|128|0+)\.0)|((255\.)(255|254|252|248|240|224|192|128|0+)(\.0+){2})|((255|254|252|248|240|224|192|128|0+)(\.0+){3}))$/,
+            hostName_IpAddress: /(^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$)|(^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9])\.)+([A-Za-z]|[A-Za-z][A-Za-z0-9-]*[A-Za-z0-9])$)/,
+            dockerReference: /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))*(:\d+)?\/)?[a-z0-9]+(([._]|__|[-]*)[a-z0-9]+)*(\/[a-z0-9]+(([._]|__|[-]*)[a-z0-9]+)*)*(:[\w][\w.-]{0,127})?(@[A-Za-z][A-Za-z0-9]*([-_+.][A-Za-z][A-Za-z0-9]*)*:[0-9a-fA-F]{32,})?$/,
+
+            getMaxLength: getMaxLength,
+            getValidationRules: getValidationRules
+        };
+
+        //
+        // Public methods
+        //
+
+        /**
+         * Provides maximum length of text that can be filled in input
+         * @param {string} path - path to field
+         * @returns {number}
+         */
+        function getMaxLength(path) {
+            return lodash.cloneDeep(lodash.get(lengths, path, lengths.default));
+        }
+
+        /**
+         * Returns the list of validation rules for `type`, optionally appending provided additional rules.
+         * @param {string} type - The property path to the list of validation rules.
+         * @param {Array.<Object>} [additionalRules] - Additional rules to append.
+         * @returns {Array.<Object>} the rule list of type `type` with `additionalRules` appended to it if provided.
+         */
+        function getValidationRules(type, additionalRules) {
+            return lodash.chain(validationRules)
+                .get(type)
+                .defaultTo([])
+                .cloneDeep()
+                .concat(lodash.defaultTo(additionalRules, []))
+                .value();
+        }
+
+        //
+        // Private methods
+        //
+
+        /**
+         * Converts characters string to readable format
+         * Note: converts Hyphens to En Dashes, replaces one space with comma and space,
+         *       replaces letter `s` with `spaces` word
+         * Example: 'a-z A-Z - _ *' -> 'a–z, A–Z, –, _, *'
```

Comment:
You can use JSDoc's `@example` tag like this:
```js
/**
 * @example
 * convertToLabel('a-z A-Z - _ *');
 * // => 'a–z, A–Z, –, _, *'
 */
```
If you do, please move below `@returns`